src/Services/QueuedJobService.php

```diff
@@ -334,6 +300,98 @@ public function isAtMaxJobs()
         return false;
     }
 
+    /**
+     * Using a job signature, returns the JobDescriptor ID and whether the
+     * job descriptor is new or existing
+     *
+     * @param string $signature
+     * @param QueuedJob $job
+     * @param QueuedJobDescriptor $jobDescriptor
+     * @param null|string $startAfter
+     * @return int|null
+     */
+    protected function findOrMakeJobDescriptorFromSignature($signature, $job, $jobDescriptor, $startAfter)
```

Comment:
This should be made private to reduce the API surface.  There's an extension hook available if it needs to be extended.